state-sync/state-sync-v2/state-sync-driver/src/notification_handlers.rs

```diff
@@ -68,24 +69,40 @@ impl CommitNotification {
         CommitNotification::CommittedStateSnapshot(committed_states)
     }
 
-    /// Handles the commit notification by notifying mempool and the event
-    /// subscription service.
-    pub async fn handle_transaction_notification<M: MempoolNotificationSender>(
+    /// Handles the commit notification by notifying mempool, the event
+    /// subscription service and the storage service.
+    pub async fn handle_transaction_notification<
+        M: MempoolNotificationSender,
+        S: StorageServiceNotificationSender,
+    >(
         events: Vec<ContractEvent>,
         transactions: Vec<Transaction>,
         latest_synced_version: Version,
         latest_synced_ledger_info: LedgerInfoWithSignatures,
         mut mempool_notification_handler: MempoolNotificationHandler<M>,
         event_subscription_service: Arc<Mutex<EventSubscriptionService>>,
+        mut storage_service_notification_handler: StorageServiceNotificationHandler<S>,
     ) -> Result<(), Error> {
-        // Notify mempool of the committed transactions
+        // Notify the storage service of the committed transactions
+        let blockchain_timestamp_usecs = latest_synced_ledger_info.ledger_info().timestamp_usecs();
         debug!(
             LogSchema::new(LogEntry::NotificationHandler).message(&format!(
-                "Notifying mempool of transactions at version: {:?}",
-                latest_synced_version
+                "Notifying the storage service of transactions at version: {:?}, timestamp: {:?}",
+                latest_synced_version, blockchain_timestamp_usecs
             ))
         );
+        storage_service_notification_handler
+            .notify_storage_service_of_committed_transactions(latest_synced_version)
+            .await?;
+
+        // Notify mempool of the committed transactions
         let blockchain_timestamp_usecs = latest_synced_ledger_info.ledger_info().timestamp_usecs();
```

Comment:
This can be removed because it's reading the same thing?

Reply:
Copy/pasta, thanks 😄